sklearn/ensemble/forest.py

```diff
@@ -374,18 +374,17 @@ def feature_importances_(self):
         return sum(all_importances) / len(self.estimators_)
 
 
-# these next two are utility functions for joblib's Parallel. Ideally
-# they would be defined locally in ForestClassifier or ForestRegressor, but
-# joblib complains that it cannot Pickle them when placed there.
-
-def _run_estimator(f, X, out):
-    out += f(X, check_input=False)
-
-
-def _run_estimator2(f, X, out):
-    all_proba = f(X, check_input=False)
-    for i in range(len(out)):
-        out[i] += all_proba[i]
+# This is a utility function for joblib's Parallel. It can't go locally in
+# ForestClassifier or ForestRegressor, because joblib complains that it cannot
+# pickle it when placed there.
+
+def accumulate_prediction(predict, X, out):
+    prediction = predict(X, check_input=False)
+    if len(out) == 1:
```

Comment:
I don't have a strong opinion on any of the solution.
However, using `if isinstance()` is straight forward regarding the aim of the `if` statement.
With your solution, I think that you would need a comment preceding the `if` statement for the sake of clarity.